span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java

```diff
@@ -277,4 +285,28 @@ private Event buildEvent(
 
     return eventBuilder.build();
   }
+
+  private void logSpanConversion(Span jaegerSpan, RawSpan rawSpan) {
+    if (LOG.isDebugEnabled()) {
+      try {
+        LOG.debug("Converted Jaeger span: {} to rawSpan: {} ", jaegerSpan, convertToJsonString(rawSpan, rawSpan.getSchema()));
+      } catch (IOException e) {
+        LOG.warn("An exception occurred while converting avro to JSON string", e);
+      }
+    }
+  }
+
+  // We should have a small library for useful short methods like this one such as this.
+  public static <T extends SpecificRecordBase> String convertToJsonString(T object, Schema schema)
+      throws IOException {
+    JsonEncoder encoder;
+    try (ByteArrayOutputStream output = new ByteArrayOutputStream()) {
+      DatumWriter<T> writer = new SpecificDatumWriter<>(schema);
+      encoder = EncoderFactory.get().jsonEncoder(schema, output, false);
```

Comment:
what happens to the binary fields like `trace_id` as part of this logging?

Reply:
The encoder just encodes as is. It will spit out '?' for unrecognized characters. Which is ok. We dont care about the value when debugging usually. We just care about the consistency of the id values.